### PR TITLE
169 comment example code

### PR DESCRIPTION
**Pull request by da-kami**
Wanted to give this a try to get early feedback. 

Once @D4nte 's PRs are merged I will update it.

### CODE REVIEW

File: new_project/examples/separate_apps/src/maker.ts [hunk 1]
```diff
@@ -40,46 +57,55 @@ function createOrder(): Order {
...
     const invitationDetails = `http://localhost:2318/orders/ETH-BTC`;
     console.log(`Waiting for someone taking my order at: ${invitationDetails}`);
 
+    // Wait for a taker to accept the order and send a swap request through the comit network daemon (cnd).
     let swapHandle;
+    // This loop runs until a swap request was sent from the taker to the maker
+    // and a swap is waiting to be processed on the maker's side.
     while (!swapHandle) {
         await new Promise(r => setTimeout(r, 1000));
+        // Listen for incoming swaps in the comit node daemon (cnd) of the maker.
```
Comment (D4nte) [inline new_project/examples/separate_apps/src/maker.ts]:
```suggestion
        // Check for incoming swaps in the comit node daemon (cnd) of the maker.
```

File: new_project/examples/separate_apps/src/maker.ts [hunk 2]
```diff
@@ -89,19 +115,12 @@ function createOrder(): Order {
...
 
-    // only accept a request if it fits to the created order above
-    if (isValid(swapParams, order)) {
-        console.log("Requested order is invalid");
-        await swapHandle.decline(actionConfig);
-        process.exit();
-    }
-    console.log("Requested order is still valid");
-    await swapHandle.accept(actionConfig);
+    // Define how often and how long the comit-js-sdk should try to execute the accept/decline, fund and redeem action.
```
Comment (D4nte) [inline new_project/examples/separate_apps/src/maker.ts]:
```suggestion
    // Define how often and how long the COMIT SDK should try to fetch the accept/decline, fund and redeem action details to then execute them using a wallet.
```

File: new_project/examples/separate_apps/src/maker.ts [hunk 3]
```diff
@@ -110,35 +129,62 @@ function createOrder(): Order {
...
         "Bitcoin HTLC funded! TXID: ",
+
+        // -- FUND --
+        // Wait for the successful execution of the funding transaction of the maker.
+        //
+        // -- Execution Details: --
+        // The maker's fund transaction will only be executed after the maker's comit network daemon (cnd)
+        // has detected the funding transaction of the taker. (The taker funds first.)
+        //
+        // This future will thus resolve once:
```
Comment (D4nte) [inline new_project/examples/separate_apps/src/maker.ts]:
```suggestion
        // This promise will thus resolve once:
```

File: new_project/examples/separate_apps/src/taker.ts [hunk 4]
```diff
@@ -5,13 +5,27 @@ import {
...
-import { checkEnvFile, startClient } from "./lib";
+import { startClient } from "./lib";
 
+/**
+ * This executable function represents the taker side during a trade.
+ * A trade consists of two phases: negotiation and execution.
+ *
+ * During the negotiation phase the taker retrieves orders that the maker publishes.
+ * The taker then has to decide if he wants to do a swap according to the order (i.e. take the order).
+ * Once the negotiation is is over (i.e. the taker has accepted the order) the execution of the swap commences.
```
Comment (D4nte) [inline new_project/examples/separate_apps/src/taker.ts]:
```suggestion
 * Once the negotiation is over (i.e. the taker has accepted the order) the execution of the swap commences.
```

File: new_project/examples/separate_apps/src/taker.ts [hunk 5]
```diff
@@ -5,13 +5,27 @@ import {
...
+ *
+ * -- Execution details: --
+ * Most of the logic of the swap execution is done by the comit-js-sdk. The example tells the ComitClient that
```
Comment (D4nte) [inline new_project/examples/separate_apps/src/taker.ts]:
```suggestion
 * Most of the logic of the swap execution is done by the COMIT SDK. The example tells the ComitClient that
```

File: new_project/examples/separate_apps/src/taker.ts [hunk 6]
```diff
@@ -39,19 +60,17 @@ import { checkEnvFile, startClient } from "./lib";
...
-    const swapParameters = swapMessage.properties!.parameters;
-    const ether = formatEther(order.ask.amount);
-    const bitcoin = toBitcoin(order.bid.amount);
     console.log(
         `Received latest order details: %s:%s for a rate of %d:%d`,
         order.ask.asset,
         order.bid.asset,
-        ether,
-        bitcoin
+        formatEther(order.ask.amount),
```
Comment (D4nte) [inline new_project/examples/separate_apps/src/taker.ts]: merge conflicts :)

File: new_project/examples/separate_apps/src/taker.ts [hunk 7]
```diff
@@ -39,19 +60,17 @@ import { checkEnvFile, startClient } from "./lib";
...
+        toBitcoin(order.bid.amount)
     );
 
-    const actionConfig = { timeout: 100000, tryInterval: 1000 };
+    // Retrieve the details (properties) of the swap
+    const swapMessage = await swap.getEntity();
```
Comment (D4nte) [inline new_project/examples/separate_apps/src/taker.ts]: so much merge conflicts

File: new_project/examples/separate_apps/src/taker.ts [hunk 8]
```diff
@@ -61,21 +80,61 @@ import { checkEnvFile, startClient } from "./lib";
...
+    console.log(
+        "Ethereum HTLC funded! TXID: ",
 
+        // -- FUND --
+        // Wait for the successful execution of the funding transaction of the taker.
+        //
+        // -- Execution Details: --
+        // The taker is the first one to fund, thus this is the first transaction sent.
+        //
+        // This future will thus resolve once:
```
Comment (da-kami) [inline new_project/examples/separate_apps/src/taker.ts]:
```suggestion
        // This promise will thus resolve once:
```

File: new_project/examples/separate_apps/src/taker.ts [hunk 9]
```diff
@@ -61,21 +80,61 @@ import { checkEnvFile, startClient } from "./lib";
...
+        "Bitcoin redeemed! TXID: ",
+
+        // -- REDEEM --
+        // Wait for the successful execution of the redeem transaction of the taker.
+        //
+        // -- Execution Details: --
+        // The takers's redeem transaction will only be executed after the taker's comit network daemon (cnd)
+        // has detected the fund transaction of the maker.
+        //
+        // This future will thus resolve once:
```
Comment (da-kami) [inline new_project/examples/separate_apps/src/taker.ts]:
```suggestion
        // This promise will thus resolve once:
```

File: new_project/examples/separate_apps/src/maker.ts [hunk 10]
```diff
@@ -110,35 +129,62 @@ function createOrder(): Order {
...
         "Ether redeemed! TXID: ",
+
+        // -- REDEEM --
+        // Wait for the successful execution of the redeem transaction of the maker.
+        //
+        // -- Execution Details: --
+        // The maker's redeem transaction will only be executed after the maker's comit network daemon (cnd)
+        // has detected the redeem transaction of the taker. (The taker redeems first.)
+        //
+        // This future will thus resolve once:
```
Comment (da-kami) [inline new_project/examples/separate_apps/src/maker.ts]:
```suggestion
        // This promise will thus resolve once:
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Changed
 - `separate_apps` example now uses the negotiation protocol provided by comit-sdk.
```
Comment (D4nte) [inline CHANGELOG.md]: Hum, I am guessing it needs rebase? We shoud see `0.5.0` here.

File: new_project/examples/separate_apps/README.md [hunk 12]
```diff
@@ -1,7 +1,20 @@
-# Example for swapping Bitcoin for Ether (separate processes) 
+# Example for swapping Ether for Bitcoin (separate processes)
```
Comment (D4nte) [inline new_project/examples/separate_apps/README.md]: The order does not matter as it depends on the pov (taker vs maker).

File: new_project/examples/separate_apps/README.md [hunk 13]
```diff
@@ -1,7 +1,20 @@
-# Example for swapping Bitcoin for Ether (separate processes) 
+# Example for swapping Ether for Bitcoin (separate processes) 
 
 An example project that shows how to swap Bitcoin for Ether with COMIT.
 
+In this example the taker and maker are split up in separate executable functions.
```
Comment (D4nte) [inline new_project/examples/separate_apps/README.md]:
```suggestion
In this example the taker and maker are split up in separate executable files.
```

File: new_project/examples/separate_apps/README.md [hunk 14]
```diff
@@ -1,7 +1,20 @@
...
+Upon executing the maker and taker apps the execution will bounce between the maker and taker, making the example more realistic.
```
Comment (D4nte) [inline new_project/examples/separate_apps/README.md]:
```suggestion
Upon executing the maker and taker apps the execution will bounce between the maker and taker, making the example closer to a production app.
```
🙃

File: new_project/examples/separate_apps/README.md [hunk 15]
```diff
@@ -1,7 +1,20 @@
...
+
+The example includes the simple negotiation protocol provided by the COMIT SDK.
+
+The example sets off at the point where the maker has already published an order for the taker.
+The example then consists of 5 steps:
```
Comment (D4nte) [inline new_project/examples/separate_apps/README.md]:
```suggestion
It then consists of 5 steps:
```

File: new_project/examples/separate_apps/README.md [hunk 16]
```diff
@@ -1,7 +1,20 @@
...
+# Example for swapping Ether for Bitcoin (separate processes) 
 
 An example project that shows how to swap Bitcoin for Ether with COMIT.
 
+In this example the taker and maker are split up in separate executable functions.
+Upon executing the maker and taker apps the execution will bounce between the maker and taker, making the example more realistic.
+
+The example includes the simple negotiation protocol provided by the COMIT SDK.
+
+The example sets off at the point where the maker has already published an order for the taker.
```
Comment (D4nte) [inline new_project/examples/separate_apps/README.md]:
```suggestion
It sets off at the point where the maker has already published an order for the taker.
```

File: new_project/examples/separate_apps/src/maker.ts [hunk 17]
```diff
@@ -8,31 +8,25 @@ import { formatEther } from "ethers/utils";
...
-    };
-}
-
+import { startClient } from "./lib";
+
+/**
+ * This executable function represents the maker side during a trade.
+ * A trade consists of two phases: negotiation and execution.
+ *
+ * During the negotiation phase the maker publishes and order that the taker can take.
```
Comment (D4nte) [inline new_project/examples/separate_apps/src/maker.ts]:
```suggestion
 * During the negotiation phase the maker publishes an order that the taker can take.
```

File: new_project/examples/separate_apps/src/maker.ts [hunk 18]
```diff
@@ -41,48 +35,72 @@ function createOrder(): Order {
...
+            asset: "bitcoin",
+            ledger: "bitcoin",
+        },
+    };
+
+    // Publish the order so the taker can take it.
     makerNegotiator.addOrder(order);
 
+    // Let the world know that there is an order available.
+    // In a real-world application this information could be shared on e.g. social media.
```
Comment (D4nte) [inline new_project/examples/separate_apps/src/maker.ts]:
```suggestion
    // In a real-world application this information could be shared publicly, e.g. on social medias.
```

File: new_project/examples/separate_apps/src/maker.ts [hunk 19]
```diff
@@ -92,51 +110,73 @@ function createOrder(): Order {
         });
     }
 
+    // Retrieve the details (properties) of the swap
     const swap = await swapHandle.fetchDetails();
     const swapParams = swap.properties!.parameters;
 
-    // only accept a request if it fits to the created order above
```
Comment (D4nte) [inline new_project/examples/separate_apps/src/maker.ts]: Sorry, merge conflict incoming as I am removing this in my patch.

File: new_project/examples/separate_apps/src/taker.ts [hunk 20]
```diff
@@ -22,11 +35,17 @@ import { checkEnvFile, startClient } from "./lib";
...
+    readLineSync.question("1. Ready to accept and order from the maker?");
 
+    // Initialize the taker negotiator that defines the negotiation phase of the trade.
+    // The taker negotiator manages retrieving orders from the maker and deciding if they are acceptable for the taker.
+    // Once an order was taken by a taker the negotiator hands over the order and execution parameters to the
+    // execution phase.
     const takerNegotiator = new TakerNegotiator(taker.comitClient);
     const makerClient = new MakerClient("http://localhost:2318/");
 
+    // TODO: Add comments once Franck's PR is merged
```
Comment (D4nte) [inline new_project/examples/separate_apps/src/taker.ts]: Don't forget this todo.

**Comment by da-kami** [comment]
will create a new PR shortly, that fixes the merge/rebase problem

**Pull request closed without merging**